### PR TITLE
S3 storage driver

### PR DESCRIPTION
**Pull request by csansoon**
## Describe your changes

WIP!!

## Checklist before requesting a review
- [ ] I have added thorough tests
- [ ] I have updated the documentation if necessary
- [x] I have added a human-readable description of the changes for the release notes
- [ ] I have included a recorded video capture of the feature manually tested

### CODE REVIEW

File: packages/connectors/duckdb/src/index.ts [hunk 1]
```diff
@@ -134,6 +145,25 @@ export default class DuckdbConnector extends BaseConnector<ConnectionParams> {
...
+
+  private async connectS3(credentials: ConnectionParams['s3']): Promise<void> {
+    if (!credentials) return
+    const { accessKeyId, secretAccessKey, region } = credentials
+    const query = `
+INSTALL httpfs;
+LOAD httpfs;
+CREATE SECRET (
+  TYPE S3,
+  KEY_ID ${JSON.stringify(accessKeyId)},
```
Comment (andresgutgon) [inline packages/connectors/duckdb/src/index.ts:161]: Do you need to stringify this? Is already a string no?

Comment (csansoon) [reply packages/connectors/duckdb/src/index.ts:161]: This correctly formats the value, wrapping it in colons and escaping whatever needs escaping.

File: packages/storage_driver/src/drivers/s3/index.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    upload
+      .done()
+      .then(() => pass.emit(CUSTOM_CLOSE_EVENT))
+      .catch((error) => pass.emit('error', error))
+
+    return pass
+  }
+
+  async createReadStream(filepath: string): Promise<Readable> {
```
Comment (andresgutgon) [inline packages/storage_driver/src/drivers/s3/index.ts:163]: This should be a private method no?

Comment (csansoon) [reply packages/storage_driver/src/drivers/s3/index.ts]: No, but it's not actually being used atm. I'll remove it

**Review by andresgutgon — APPROVED** [review]

**Pull request merged**